packages/jest-worker/src/workers/__tests__/threadChild.test.ts

```diff
@@ -166,250 +163,205 @@ it('lazily requires the file', () => {
     'Number',
     void 0,
     void 0,
     412,
   ]);
 
-  thread.emit('message', [
+  messagePort.emit('message', [
     CHILD_MESSAGE_CALL,
-    true, // Not really used here, but for flow type purity.
+    true,
     'fooThrowsAnErrorWithExtraProperties',
     [],
   ]);
 
-  expect(thread.postMessage.mock.calls[3][0]).toEqual([
+  expect(jest.mocked(messagePort.postMessage).mock.calls[3][0]).toEqual([
     PARENT_MESSAGE_CLIENT_ERROR,
-    'ReferenceError',
-    'Booo extended',
+    'MockExtendedError',
+    'Boo extended',
     mockExtendedError.stack,
     {baz: 123, qux: 456},
   ]);
 
-  thread.emit('message', [
-    CHILD_MESSAGE_CALL,
-    true, // Not really used here, but for flow type purity.
-    'fooThrowsNull',
-    [],
-  ]);
+  messagePort.emit('message', [CHILD_MESSAGE_CALL, true, 'fooThrowsNull', []]);
 
-  expect(thread.postMessage.mock.calls[4][0][0]).toBe(
+  expect(jest.mocked(messagePort.postMessage).mock.calls[4][0][0]).toBe(
     PARENT_MESSAGE_CLIENT_ERROR,
   );
-  expect(thread.postMessage.mock.calls[4][0][1]).toBe('Error');
-  expect(thread.postMessage.mock.calls[4][0][2]).toBe(
+  expect(jest.mocked(messagePort.postMessage).mock.calls[4][0][1]).toBe(
+    'Error',
+  );
+  expect(jest.mocked(messagePort.postMessage).mock.calls[4][0][2]).toBe(
     '"null" or "undefined" thrown',
   );
 
-  expect(thread.postMessage).toHaveBeenCalledTimes(5);
+  expect(messagePort.postMessage).toHaveBeenCalledTimes(5);
 });
 
 it('returns results when it gets resolved if function is asynchronous', async () => {
-  thread.emit('message', [
+  messagePort.emit('message', [
     CHILD_MESSAGE_INITIALIZE,
-    true, // Not really used here, but for flow type purity.
+    true,
     './my-fancy-worker',
   ]);
 
-  thread.emit('message', [
+  messagePort.emit('message', [
     CHILD_MESSAGE_CALL,
-    true, // Not really used here, but for flow type purity.
+    true,
     'fooPromiseWorks',
     [],
   ]);
 
   await sleep(10);
 
-  expect(thread.postMessage.mock.calls[0][0]).toEqual([
+  expect(jest.mocked(messagePort.postMessage).mock.calls[0][0]).toEqual([
     PARENT_MESSAGE_OK,
     1989,
   ]);
 
-  thread.emit('message', [
+  messagePort.emit('message', [
     CHILD_MESSAGE_CALL,
-    true, // Not really used here, but for flow type purity.
+    true,
     'fooPromiseThrows',
     [],
   ]);
 
   await sleep(10);
 
-  expect(thread.postMessage.mock.calls[1][0]).toEqual([
+  expect(jest.mocked(messagePort.postMessage).mock.calls[1][0]).toEqual([
     PARENT_MESSAGE_CLIENT_ERROR,
     'TypeError',
-    'Booo',
+    'Boo',
     mockError.stack,
     {},
   ]);
 
-  expect(thread.postMessage).toHaveBeenCalledTimes(2);
+  expect(messagePort.postMessage).toHaveBeenCalledTimes(2);
 });
 
 it('calls the main module if the method call is "default"', () => {
-  thread.emit('message', [
+  messagePort.emit('message', [
     CHILD_MESSAGE_INITIALIZE,
-    true, // Not really used here, but for flow type purity.
+    true,
```

Comment:
`true` must stay here. Tests are failing if I remove it. Only comments are redundant.